enterprise/coderd/prebuilds/reconcile.go

```diff
@@ -0,0 +1,511 @@
+package prebuilds
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"math"
+	"sync/atomic"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+
+	"github.com/coder/quartz"
+
+	"github.com/coder/coder/v2/coderd/audit"
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/database/dbauthz"
+	"github.com/coder/coder/v2/coderd/database/provisionerjobs"
+	"github.com/coder/coder/v2/coderd/database/pubsub"
+	"github.com/coder/coder/v2/coderd/prebuilds"
+	"github.com/coder/coder/v2/coderd/rbac"
+	"github.com/coder/coder/v2/coderd/rbac/policy"
+	"github.com/coder/coder/v2/coderd/wsbuilder"
+	"github.com/coder/coder/v2/codersdk"
+
+	"cdr.dev/slog"
+
+	"github.com/google/uuid"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/xerrors"
+)
+
+type StoreReconciler struct {
+	store  database.Store
+	cfg    codersdk.PrebuildsConfig
+	pubsub pubsub.Pubsub
+	logger slog.Logger
+	clock  quartz.Clock
+
+	cancelFn context.CancelCauseFunc
+	stopped  atomic.Bool
+	done     chan struct{}
+}
+
+var _ prebuilds.ReconciliationOrchestrator = &StoreReconciler{}
+
+func NewStoreReconciler(store database.Store, ps pubsub.Pubsub, cfg codersdk.PrebuildsConfig, logger slog.Logger, clock quartz.Clock) *StoreReconciler {
```

Comment:
Throughout this file, pleas pay attention to line lenth, max 120 chars.

Reply:
Fixed most of the things.